app/components/multi-select-dropdown/multi-select-dropdown.component.ts

```diff
@@ -0,0 +1,100 @@
+import angular from 'angular';
+import template from './multi-select-dropdown.component.html';
+import _ from 'lodash';
+
+type DropdownDataObj = {
+  id?: number;
+  name?: string;
+  value?: string;
+  group?: string;
+  label?: string;
+};
+
+class MultiSelectDropdown {
+  options!: DropdownDataObj[];
+  selectedOptions!: any[];
+  smartButtonText?: string;
+  buttonText?: string;
+  placeholderText?: string;
+  overrideOption?: DropdownDataObj;
+  onOptionChanged!: any;
+  maxWidth?: string;
+  smartTextWidth?: string;
+
+  async $onInit() {
+      this.smartButtonText = this.buttonText ?? '';
+      this.maxWidth = this.maxWidth ?? '260px';
+      this.smartTextWidth = this.smartTextWidth ?? '190px';
+  }
+
+  public onOptionSelection(option) {
+      const isOptionSelected = _.findIndex(
+          this.selectedOptions,
+          (o) => o.label === option.label
+      );
+
+      /* if override option is chosen, clear selectedOptions list and just add that one
+      if anything else gets selected, remove override option */
```

Comment:
super picky, but could you space this so it's on two lines and reword a bit? My brain had a hard time interpreting it. 

Something like

/* 
    If override option is chosen, clear selectedOptions list and just add the overrideOption
    If anything else gets selected, remove override option 
*/